install/pre-install-payload/scripts/reqs-deploy.sh

```diff
@@ -260,13 +266,13 @@ function main() {
 		print_help && die ""
 	fi
 
-	if [ ! -f "${containerd_config}" ]; then
+	set_container_engine
+
+	if [[ ${container_engine} = "containerd" && ! -f "${containerd_config}" ]]; then
 		mkdir -p /etc/containerd
 		containerd config default > /etc/containerd/config.toml
 	fi
 
-	set_container_engine
```

Comment:
Not sure why this is changing the order of set_container_engine and default config generation, but as this function simply sets a variable it shouldn't matter (but it'd be purer to keep the order ;-) )

Reply:
the old if was not comparing the container_engine variable. Since now we are checking for that, we need to call set_container_engine before the if.